azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/artifacthandler/FTPArtifactHandlerImpl.java

```diff
@@ -32,34 +33,47 @@ public FTPArtifactHandlerImpl(final T mojo) {
     }
 
     protected String getDeploymentStageDirectory() {
-        final String outputFolder = this.mojo.getPluginName().replaceAll(MAVEN_PLUGIN_POSTFIX, "");
-        return Paths.get(mojo.getBuildDirectoryAbsolutePath(), outputFolder, this.mojo.getAppName()).toString();
+        return Paths.get(mojo.getBuildDirectoryAbsolutePath(), this.mojo.getAppName()).toString();
     }
 
     @Override
     public void publish(DeployTarget target) throws IOException, MojoExecutionException {
+        if (target.getApp() instanceof WebApp) {
+            prepareResources();
+            assureStageDirectoryNotEmpty();
```

Comment:
Any reason to add this check method here?

Reply:
function app does not need copy resources to stage directory

Reply:
I mean the method `assureStageDirectoryNotEmpty()`

Reply:
jar deploy does not need copy resources neither, it just prepare the jar file to stage directory itself. 

This change use function prepareResources() just do copy resources (if any), and leave resources check to function assureStageDirectoryNotEmpty()

Reply:
OK, that's fine. But the error handling may need to be improved.

The string `NO_RESOURCES` contains two scenario:
- No resources were specified in pom.xml
- No resources were copied to stage directory

A better way is to separate these two conditions, and let the user know what's happening.

And for `No resources were specified in pom.xml` condition, do we need to throw an error or just a warning to tell the user the deployment is skipped?